config/v1/types_cluster_version.go

```diff
@@ -290,6 +305,100 @@ type Release struct {
 // availableUpdates field is accurate and recent.
 const RetrievedUpdates ClusterStatusConditionType = "RetrievedUpdates"
 
+// ConditionalUpdate represents an update which is recommended to some
+// clusters on the version the current cluster is reconciling, but which
+// may not be recommended for the current cluster.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdate struct {
+	// release is the target of the update.
+	// +required
+	Release Release `json:"release"`
+
+	// risks represents the range of issues associated with
+	// updating to the target release. The cluster-version
+	// operator will evaluate all entries, and only recommend the
+	// update if there is at least one entry and all entries
+	// recommend the update.
+	// +kubebuilder:validation:MinItems=1
+	// +required
+	Risks []ConditionalUpdateRisk `json:"risks"`
```

Comment:
needs patch/list type keys for SSA

Reply:
I'm a bit fuzzy on how these work, but I've taken a stab with 37280ee5 -> 866951f38, although [now CI is giving me some trouble][1].

[1]: https://github.com/openshift/api/pull/1011#issuecomment-925371910

Reply:
Aha, here we go [tucked in the output a hundred or so lines from the end][1]:

```
The CustomResourceDefinition "clusterversions.config.openshift.io" is invalid: spec.validation.openAPIV3Schema.properties[status].properties[conditionalUpdates].items.properties[risks].items.properties[name].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
```

But [`name` is `+required`][2]...  As of 866951f38bfc:

```console
$ yaml2json <config/v1/0000_00_cluster-version-operator_01_clusterversion.crd.yaml | jq -r '.spec.versions[].schema.openAPIV3Schema.properties.status.properties.conditionalUpdates.items.properties.
conditions.items.required[]'
lastTransitionTime
message
reason
status
type
```

But:

```console
$ yaml2json <config/v1/0000_00_cluster-version-operator_01_clusterversion.crd.yaml | jq -r '.spec.versions[].schema.openAPIV3Schema.properties.status.properties.conditionalUpdates.items.properties.risks.items | keys[]'
description
properties
type
```

Perhaps `+required` needs to come before the `+kubebuilder...`?  I pushed 866951f3 -> 6f1954491fa97f to try, but [it didn't help][3].

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_api/1011/pull-ci-openshift-api-master-verify/1440808600376184832#1:build-log.txt%3A1002
[2]: https://github.com/openshift/api/pull/1011/commits/866951f38bfcc0f919547120516cd986b746eb0f#diff-1b89ebfc7018fedcaa25a406ba7d376d0aa0f025a416bd286238899b6ab7ba91R358
[3]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_api/1011/pull-ci-openshift-api-master-verify/1440813791037100032#1:build-log.txt%3A1002

Reply:
Ahh, I seem to have bumped into #376.  I dunno why we're keeping the old `+required` around to confuse folks, but I've added explicit `+kubebuilder:validation:Required` in 6f195449 -> c767ec72.